src/ray/raylet/worker_pool.cc

```diff
@@ -557,6 +557,13 @@ void WorkerPool::MonitorStartingWorkerProcess(const Process &proc,
                   ? "The process is still alive, probably it's hanging during start."
                   : "The process is dead, probably it crashed during start.");
 
+      if (first_job_send_register_client_reply_to_driver_) {
+        // We should also warn that since some workers are hanging, the first job callback
+        // if present might not be called yet. This might cause the first job to be
+        // still hanging.
+        RAY_LOG(WARNING) << "The first job callback is not called yet.";
```

Comment:
I think this log message doesn't make much sense. Maybe write more high level? Also is this called every time for each process? Maybe consider using RAY_LOG_EVERY_MS (every 10 seconds?)? 

"""
The first job of ID X registration hangs because X workers cannot be started on time due to timeout. 
"""

"""